src/mono/wasm/runtime/library_mono.js

```diff
@@ -634,11 +633,12 @@ var MonoSupportLib = {
 					if (ctx.tracing)
 						console.log ("MONO_WASM: Creating file '" + fileName + "' in directory '" + parentDirectory + "'");
 
-					var fileRet = ctx.createDataFile (
-						parentDirectory, fileName,
-						bytes, true /* canRead */, true /* canWrite */, true /* canOwn */
-					);
-
+					if (!this.mono_wasm_load_data_archive (bytes, parentDirectory)) {
```

Comment:
I feel like there's a risk of getting mystery breakage here if a file happens to begin with the magic number, but it's probably fine for now.

Reply:
I agree in principle and and I'm fine with adding an "behavior" but I think if should be a content-type instead of of a behavior and the loading function should be extensible off the the type with the pre defined types having default handlers.  I think this is a good step towards that goal.

Reply:
To add more than sufficient uniqueness we could verify that the manifest parses as json and that that the length on the manifest is less that then length of the data*

* the ability to add another archive at the excess of the previous archive has proven valuable in the past